### ISSUE TITLE
Curator unable to delete hidden directories (prefix with dot in linux)

### ISSUE THREAD

**Issue by mtunstill** [elastic/curator#877]
We've encountered an issue with Curator 3.4.0 where it won't fail, even though it's unable to delete (or seems to ignore) hidden directories in Linux when they start with a dot. Curator will clear-up the metadata, but not the data in the indices directory. This is hurting us on .marvel-* indices especially where the default is '.marvel-%Y.%m.%d'.

This is against a ES 1.7.2 cluster (I'm aware this is now EOL).

**Comment by mtunstill** [comment]
Just to add some more info, this is the command we're using... (even following the docs where you need the dash)

/usr/bin/curator --host hostname --http_auth curator:password --port 9200 --use_ssl --certificate /opt/curator/certs/root.cer delete indices --older-than 90 --time-unit days --timestring \%Y.\%m.\%d --prefix .marvel-

**Comment by untergeek** [comment]
This has_never been a problem before.  I'm not sure what you're up against, but that command has worked for many others (myself included) when using Curator 3.x.

The issue at hand here seems to be revealed when you say that

> Curator will clear-up the metadata, but not the data in the indices directory.

This is revealing because Curator only issues API calls to Elasticsearch and returns a response.  It seems that Curator is sending the API call, and Elasticsearch is saying, "I got this," and responding that it is deleting the data.  If Elasticsearch is not deleting the data, that is not going to be Curator's fault, unfortunately.  Curator behaves as if you ran:

```
curl -XDELETE http://hostname:port/.marvel-2016.01.31
```

multiple times, one line for each index.

To recap, if Elasticsearch is deleting _anything_—including metadata—it suggests that Curator made the API call to Elasticsearch successfully.  If Elasticsearch failed to complete the deletion, I don't know what to tell you.

Do you have more in the logs?  Debug from Curator, and/or from Elasticsearch (the client and/or the elected master node)?

**Comment by mtunstill** [comment]
Hi @untergeek,

I wasn't sure if Curator has any other built in functions outside of ES remit to amend data via its APIs. It's a very odd one as we're applying the same command lines against an older version of Curator in a different environment (v3.0.3) and that has no issues as far as I can tell. The main difference is that ES has shield installed where we're experiencing the issue.

I'll dig deeper into the logs and if it's something I can't figure out with the information you've given me, I'll post them here. The other thing is that the backup directory is an NFS mount hosted on a SAN device.

I'd of thought Curator would fail if ES was unable to delete the indices (as I've seen before).

**Comment by untergeek** [comment]
If you're running Shield, it probably has something to do with that.  In the earlier versions, dot-prefixed index names were protected, if I recall correctly.  I'm not sure what can be done to address that.

**Comment by untergeek** [comment]
In any case, it's not Curator itself that can correct the problem.

**Comment by untergeek** [comment]
Closing.
